src/google/cloud/happybase/batch.py

```diff
@@ -162,9 +162,11 @@ def put(self, row, data, wal=_WAL_SENTINEL):
         column_pairs = _get_column_pairs(six.iterkeys(data),
                                          require_qualifier=True)
 
-        for column_family_id, column_qualifier in column_pairs:
-            value = data[(column_family_id + ':' +
-                          column_qualifier).encode('utf-8')]
+        # use key that was passed. Reconstructing it can cause it to not
+        # be found if there is an encoding difference.
+        for key, column_pair in zip(six.iterkeys(data), column_pairs):
+            column_family_id, column_qualifier = column_pair
+            value = data[key]
```

Comment:
Column qualifiers [*must* be bytes](https://cloud.google.com/bigtable/docs/reference/data/rpc/google.bigtable.v2#google.bigtable.v2.Column).  The alternatives would be:

- Raise an exception if the caller passes in text rather than bytes.
- Define a `default_encoding` option for `Batch` which we use, if set, to encode column qualifiers passed as text.

Reply:
As written, the encoding is made to be bytes. What this fixes is that `data[key]` != `data[key.encode('utf-8')]` which was causing lookup failures.

Reply:
The other proposal would be to throw if a user attempts to use unencoded column_qualifiers. Bigtable, as shown in @tseaver  link above requires bytes. And the sys test for this shows the oddness that occurs with us trying to auto encode.

Reply:
I have made the change to require column names, which include qualifiers, to be presented as bytes.